components/eamxx/src/share/field/field_impl.hpp

```diff
@@ -205,6 +203,56 @@ auto Field::get_strided_view () const
   return DstView(get_ND_view<HD,DstValueType,1>());
 }
 
+// NOTE: multi-slicing a view is only supported for strided view return type
+// TODO: N doesn't actually need to be passed in, since the subview must have
+// the same rank as its parent
+template <typename DT, int N, HostOrDevice HD>
+auto Field::get_multi_sliced_view() const
```

Comment:
I don't see this function called anywhere, which means the user must be aware that this field is a mutli-slice subfield of another, and call this function manually. That's probably asking a bit too much to the user. It would be nice if `get_strided_view` called this method internally (or maybe just add bits of this method to `get_strided_view`). 

Right now, `get_strided_view` _assumes_ that the output view is 1d, since we only have the need for strided views for 1d subfields. But that assumption is not really _needed_ (more of a correctness check, since we didn't expect any strided view usage for rank 2+ fields).

In short: let's make `get_strided_view` handle the multi-slice case. That will require allowing rank 2+ views for that method.

Reply:
It dawned on me that some Field utilities call either `get_view` or `get_strided_view`, depending on whether `FieldAllocProp::contiguous()` returns true or false. My previous comment becomes more important in this regard: moving this case into `get_strided_view` allows to keep the downstream code simpler, since they only have two possible methods to call: `get_view` or `get_strided_view`.

Furthermore, since multi-slice view is strided but may have rank>1, we need to add more if statements to those Field utilities. Namely, for rank 2+, those utilities need to do the same thing we do for rank-1: check if allocation is contiguous, and if so, call `get_view`, otherwise call `get_strided_view`.

Reply:
Ah, ok! That makes a lot of sense--will knock that out asap

Reply:
I believe we should be just about finished after the recent pushes. All tests passing locally for me, and  I'll need to do a rebase before merge, but otherwise let me know if this approach works for you--thanks!